types/react-relay/lib/relay-experimental/EntryPointTypes.d.ts

```diff
@@ -102,90 +91,150 @@ export type PreloadQueryStatus = Readonly<{
-    queries: JSX.LibraryManagedAttributes<TEntryPointComponent, 'queries'>;
-}>;
-
-export type ThinQueryParams<TQuery extends OperationType, TEnvironmentProviderOptions> = Readonly<{
-    environmentProviderOptions?: TEnvironmentProviderOptions | null;
-    options?: PreloadOptions | null;
-    parameters: PreloadableConcreteRequest<TQuery>;
-    variables: TQuery['variables'];
-}>;
-
-export type ThinNestedEntryPointParams<TEntryPointParams, TEntryPoint> = Readonly<{
-    entryPoint: TEntryPoint;
-    entryPointParams: TEntryPointParams;
-}>;
-
-export type ExtractQueryTypeHelper<TEnvironmentProviderOptions> = <TQuery extends OperationType>(
-    query: PreloadedQuery<TQuery>,
-) => ThinQueryParams<TQuery, TEnvironmentProviderOptions>;
-
-export type ExtractEntryPointTypeHelper<TEntryPointParams> = <TEntryPointComponent>(
-    entryPoint: PreloadedEntryPoint<TEntryPointComponent> | null | undefined,
-) =>
-    | ThinNestedEntryPointParams<TEntryPointParams, EntryPoint<TEntryPointParams, TEntryPointComponent>>
-    | null
-    | undefined;
-
-export type EntryPoint<TEntryPointParams, TEntryPointComponent> = InternalEntryPointRepresentation<
+    TPreloadedQueries extends Record<string, OperationType>,
+    TPreloadedEntryPoints extends Record<string, EntryPoint<any, any> | undefined>,
+    TRuntimeProps extends {} = {},
+    TExtraProps extends {} | null = {}
+> = ComponentType<EntryPointProps<TPreloadedQueries, TPreloadedEntryPoints, TRuntimeProps, TExtraProps>>;
+
+// Return type of `loadEntryPoint(...)`
+export type PreloadedEntryPoint<TEntryPointComponent> = TEntryPointComponent extends EntryPointComponent<
+    infer TPreloadedQueries,
+    infer TPreloadedEntryPoints,
+    infer TRuntimeProps,
+    infer TExtraProps
+>
+    ? Readonly<{
+          dispose: DisposeFn;
+          entryPoints: TPreloadedEntryPoints;
+          extraProps: TExtraProps;
+          getComponent: () => TEntryPointComponent;
+          isDisposed: boolean;
+          queries: PreloadedQueries<TPreloadedQueries>;
+          rootModuleID: string;
+      }>
+    : never;
+
+export interface ThinQueryParams<
+    TQuery extends OperationType,
+    TEnvironmentProviderOptions extends EnvironmentProviderOptions = EnvironmentProviderOptions
+> extends Readonly<{
+        /**
+         * A reference to the $Parameters file that matches the type param provided to preloadQuery type.
+         *
+         * NOTE: For now, since OSS relay-compiler doesnt support generating the $Parameters file, you must give the query
+         * wou would have given to usePreloadedQuery
+         */
+        parameters: ConcreteRequest | PreloadableConcreteRequest<TQuery>;
+        /**
+         * Variables to give to the query
+         */
+        variables: VariablesOf<TQuery>;
+        /**
+         * Any execution options to apply during network
+         */
+        options?: PreloadOptions | null;
+        environmentProviderOptions?: TEnvironmentProviderOptions | null;
```

Comment:
@edvinerikson  think we need to pass this type on as well, think this should be accessible in the `getEnvironmentProvider` cb. Except we're not passing this 👀

Reply:
Not entirely following what you mean, can you elaborate?

Reply:
I think its okay, it was just making types flow through when this is set. But its a really hard thing to type anyway. Think we can leave it for now, and revisit later.